content/tutorial/tutorial.md

```diff
@@ -1070,28 +1077,28 @@ to
+अपडेटेड काउंट्स के अलावा, यह पढ़ने वाला एक मानव शायद यह कहेगा कि हमने एलेक्सा और बेन के ऑर्डर को स्वैप किया और एलेक्सा और बेन के बीच क्लाउडिया को डाला। हालाँकि, React एक कंप्यूटर प्रोग्राम है और यह नहीं जानता कि हमारा क्या इरादा था। क्योंकि React हमारे इरादों को नहीं जान सकता की हम कोड से क्या करना चाहते हैं इसलिए हमें लिस्ट आइटम और उसके सिब्लिंग्स को अलग करने के लिए एक *key* प्रॉपर्टी हर लिस्ट आइटम पर लगाते हैं। एक विकल्प strings `alexa`,` ेben`, `claudia` का उपयोग करना होगा। यदि हम एक डेटाबेस से डेटा प्रदर्शित कर रहे थे, तो Alexa, Ben और Claudia के डेटाबेस आईडी को keys के रूप में इस्तेमाल किया जा सकता है।
 
 ```html
 <li key={user.id}>{user.name}: {user.taskCount} tasks left</li>
 ```
 
-When a list is re-rendered, React takes each list item's key and searches the previous list's items for a matching key. If the current list has a key that didn't exist before, React creates a component. If the current list is missing a key that existed in the previous list, React destroys the previous component. If two keys match, the corresponding component is moved. Keys tell React about the identity of each component which allows React to maintain state between re-renders. If a component's key changes, the component will be destroyed and re-created with a new state.
+जब एक सूची को फिर से प्रस्तुत किया जाता है, तो React प्रत्येक सूची आइटम की कुंजी लेता है और मिलान सूची के लिए पिछली सूची के आइटम को खोजता है। यदि वर्तमान सूची में एक कुंजी है जो पहले मौजूद नहीं थी, तो React एक कौम्पोनॅन्ट बनाती है। यदि वर्तमान सूची में पिछली सूची में मौजूद एक कुंजी गायब है, तो React पिछले कौम्पोनॅन्ट को नष्ट कर देती है। यदि दो कुंजी मेल खाती हैं, तो संबंधित घटक स्थानांतरित हो जाता है। कीज़ React को प्रत्येक कौम्पोनॅन्ट की पहचान के बारे में बताती है जो React को पुन: रेंडरर्स के बीच स्थिति बनाए रखने की अनुमति देता है। यदि किसी कौम्पोनॅन्ट की कुंजी बदल जाती है, तो घटक नष्ट हो जाएगा और नए state के साथ फिर से बनाया जाएगा।
 
-`key` is a special and reserved property in React (along with `ref`, a more advanced feature). When an element is created, React extracts the `key` property and stores the key directly on the returned element. Even though `key` may look like it belongs in `props`, `key` cannot be referenced using `this.props.key`. React automatically uses `key` to decide which components to update. A component cannot inquire about its `key`.
+`कुंजी` React में एक विशेष और आरक्षित संपत्ति है (`ref`, एक अधिक उन्नत सुविधा के साथ)। जब एक तत्व बनाया जाता है, तो रिएक्ट `key` संपत्ति निकालता है और कुंजी को सीधे दिए गए एलिमेंट पर संग्रहीत करता है। भले ही `key` लग सकता है, लेकिन यह `props` में है, `key` को `this.props.key` का उपयोग करके संदर्भित नहीं किया जा सकता है। React करने के लिए कौन से कौम्पोनॅन्ट को तय करना है, React स्वचालित रूप से `key` का उपयोग करता है। एक कौम्पोनॅन्ट अपने `key` के बारे में पूछताछ नहीं कर सकता।
```

Comment:
property should not be translated here

Reply:
`ref`, एक अधिक उन्नत सुविधा के साथ -> incorrect meaning of translation

Reply:
तत्व  -> element

Reply:
revisit the paragraph once :)

Reply:
```
`key` React की एक विशेष और रिजर्व्ड property है (`ref` के साथ, जो की और भी एडवांस्ड फीचर है)। जब React में एक एलिमेंट बनता है, React `key` property को निकालता है और रिटर्न्ड एलिमेंट में सीधे स्टोर करता है। भले ही लगे की `key` props के साथ है, लेकिन `key` को `this।props।key` से रिफरेन्स नहीं कर सकते। React अपने आप `key` को इस्तेमाल करता है और तय करता है की कोनसा कौम्पोनॅन्ट अपडेट करना है। कौम्पोनॅन्ट अपनी `key` के बारे में जांच नहीं कर सकता।     
```
how about this?

Reply:
cool